Submitted By: Ryan Oliver <ryan.oliver@pha.com.au>
Date: 2005-04-25
Initial Package Version: 1.15.1
Origin: Ryan Oliver
Description: Fix compilation issues with gcc4

if gcc -DHAVE_CONFIG_H -I. -I. -I.. -I../lib -I../lib -I../src    -O2 -pipe -MT genfile.o -MD -MP -MF ".deps/genfile.Tpo" -c -o genfile.o genfile.c; \
then mv -f ".deps/genfile.Tpo" ".deps/genfile.Po"; else rm -f ".deps/genfile.Tpo"; exit 1; fi
genfile.c:63: error: static declaration of 'argp_program_version' follows non-static declaration
../lib/argp.h:428: error: previous declaration of 'argp_program_version' was here
genfile.c:64: error: static declaration of 'argp_program_bug_address' follows non-static declaration
../lib/argp.h:444: error: previous declaration of 'argp_program_bug_address' was here

--- tar-1.15.1/tests/genfile.c-orig	2005-04-25 15:23:47.000000000 +1000
+++ tar-1.15.1/tests/genfile.c	2005-04-25 15:24:08.000000000 +1000
@@ -60,8 +60,8 @@
 /* Block buffer for sparse file */
 char *buffer;
 
-static const char *argp_program_version = "genfile (" PACKAGE ") " VERSION;
-static const char *argp_program_bug_address = "<" PACKAGE_BUGREPORT ">";
+const char *argp_program_version = "genfile (" PACKAGE ") " VERSION;
+const char *argp_program_bug_address = "<" PACKAGE_BUGREPORT ">";
 static char doc[] = N_("genfile generates data files for GNU paxutils test suite");
 
 static struct argp_option options[] = {
